tests/test_pricing_model.py

```diff
@@ -0,0 +1,1281 @@
+                    fee=0.06144677994914538,
+                    # We set up the problem as:
+                    #   (2 / 1.5) * (1.5 * 100_100) ** (1 - T) + (2_200_000 - d_y) ** (1 - T) = k
+                    #
+                    # Solving for d_y, we get the following calculation:
+                    #   d_y = 2_200_000 - (k - (2 / 1.5) * (1.5 * 100_100) ** (1 - T)) ** (1 / (1 - T)) = 206.14340814948082
+                    #
+                    # Note that this is slightly smaller than the without slippage value
+                    without_fee=206.14340814948082,
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 206.14340814948082 - 0.06144677994914538 = 206.08196136953168
+                    with_fee=206.08196136953168,
+                ),
+            ),
+            # A time stretch targetting 10% APY.
+            (
+                TestCaseCalcOutGivenInSuccess(
+                    in_=200,
+                    share_reserves=100_000,
+                    bond_reserves=1_000_000,
+                    token_out="pt",
+                    fee_percent=0.01,
+                    days_remaining=91.25,
+                    time_stretch_apy=0.10,
+                    share_price=2,
+                    init_share_price=1.5,
+                ),
+                # From the input, we have the following values:
+                # - T = 0.02253584403159705
+                # - p = 1.0623907066406753
+                # - k = 1_735_927.3223407117
+                TradeResult(
+                    # Using the spot price, the expected output without slippage or fees is given by:
+                    #   1.0623907066406753 * 200 = 212.47814132813505
+                    without_fee_or_slippage=212.47814132813505,
+                    # Since we are buying bonds, in_ is an amount of base and we calculate the fee using the spot price as:
+                    #   fee = 0.01 * (p - 1) * 200 = 0.1247814132813505
+                    fee=0.1247814132813505,
+                    # We set up the problem as:
+                    #   (2 / 1.5) * (1.5 * 100_100) ** (1 - T) + (2_200_000 - d_y) ** (1 - T) = k
+                    #
+                    # Solving for d_y, we get the following calculation:
+                    #   d_y = 2_200_000 - (k - (2 / 1.5) * (1.5 * 100_100) ** (1 - T)) ** (1 / (1 - T)) = 212.47551672440022
+                    #
+                    # Note that this is slightly smaller than the without slippage value
+                    without_fee=212.47551672440022,
+                    # Combining the without_fee and the fee, we calculate with_fee as:
+                    #   with_fee = 212.47551672440022 - 0.1247814132813505 = 212.35073531111888
+                    with_fee=212.35073531111888,
+                ),
+            ),
+        ]
+
+        # Test cases where token_out = "base".
+        base_out_test_cases = [
```

Comment:
Are all of these tests a repeat of the above but with `token_out = "base"`? If so, is there a simpler way to specify it? I think the below isn't exactly correct, but something like this:
```
base_out_test_cases = []
for test_case in pt_out_test_cases:
    base_test_case = test_case.copy()
    base_test_case.token_out = "base"
    base_out_test_cases.append(base_test_case)
```